### PR TITLE
#1253 Moved nan and infinites to `eo-runtime` and placed them near float

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #1253

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@Graur please take a look at this: `sources to XMIRs
Error:  net.sf.saxon.style.XSLText@26285bd0: Errors identified: Data objects can't be used directly: "org.eolang.float";`

It seems we can't do `float.nan` because `float` is data object. If we could we will be also able to do `4.2.nan` which is kind of wrong. 

Any suggestions? Maybe we should just make these objects as separate objects?

**Comment by Graur** [comment]
@maxonfjvipon Can you try to use parent `^` object instead of `float` key word inside `float` object? Maybe this error is about this mistake

**Comment by maxonfjvipon** [comment]
@Graur seems it was not only about using it inside `float` object. 
This screenshot has 50 matches with `Data objects can't be used directly: "org.eolang.float"`. Also it's 50 times when `float.something` is used it float tests. It seems to it's just not allowed at all.

image

**Comment by Graur** [comment]
@maxonfjvipon I see. Maybe we need to make `nan` as separated object? And place it near the `float`?
@yegor256 What do you think?

**Comment by Graur** [comment]
@maxonfjvipon I see. Maybe we need to make `nan` as separated object? And place it near the `float`?
@yegor256 What do you think?

**Comment by maxonfjvipon** [comment]
@yegor256 please take a look

**Comment by yegor256** [comment]
@rultor merge

**Comment by maxonfjvipon** [comment]
> @maxonfjvipon I see. Maybe we need to make `nan` as separated object? And place it near the `float`?
> 
> @yegor256 What do you think?

@yegor256 please take a look at this. Should we make these objects as separate objects since we can't place them inside `float` because it's data object?

**Comment by Graur** [comment]
@maxonfjvipon Maybe you can try to implement it and watch how it will be going?

**Comment by maxonfjvipon** [comment]
@Graur do I understand correctly that it fails because there's not enough tests?

**Comment by Graur** [comment]
@maxonfjvipon Looks like you are right

**Comment by maxonfjvipon** [comment]
@Graur could you please check this out? What can be wrong? It seems that one more test is needed but I wrote tests for all objects inside nan, +inf and -inf?

**Comment by Graur** [comment]
@maxonfjvipon Check this property: https://github.com/objectionary/eo/blob/master/eo-runtime/pom.xml#L225-L227

**Comment by maxonfjvipon** [comment]
@Graur please take a look

**Comment by Graur** [comment]
@maxonfjvipon What about tests like `0.0.eq nan`? I don't see any of them here

**Comment by maxonfjvipon** [comment]
@Graur yeah, my bad

**Comment by maxonfjvipon** [comment]
@Graur please take a look

File: eo-runtime/src/main/eo/org/eolang/nan.eo [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  [x] > minus
+    ^ > @
+
+  # Quotient of the division of $ by x
+  [x] > div
+    ^ > @
+
+  # Converts this to bytes
+  [] > as-bytes
+    7F-C0-00-00 > @
```
Comment (Graur) [inline eo-runtime/src/main/eo/org/eolang/nan.eo:92]: @maxonfjvipon Can you remind me, this value was taken from IEEE standart?

**Review by Graur — CHANGES_REQUESTED** [review]

File: eo-runtime/src/main/eo/org/eolang/nan.eo (hunk 1 above)
Comment (maxonfjvipon) [reply eo-runtime/src/main/eo/org/eolang/nan.eo:92]: @Graur yes, you can try it [here](https://www.h-schmidt.net/FloatConverter/IEEE754.html)

Comment (maxonfjvipon) [reply eo-runtime/src/main/eo/org/eolang/nan.eo:92]: @Graur yes, you can try it [here](https://www.h-schmidt.net/FloatConverter/IEEE754.html)

image

Comment (Graur) [reply eo-runtime/src/main/eo/org/eolang/nan.eo:92]: @maxonfjvipon Thanks!

**Review by Graur — APPROVED** [review]
@maxonfjvipon Now it looks better, thanks!
@yegor256 Please, have a look

**Comment by maxonfjvipon** [comment]
@yegor256 please take a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**